python/django_bolt/pagination.py

```
@@ -187,12 +187,17 @@ async def _get_queryset_count(self, queryset: Any) -> int:
-        # Check if queryset has acount (async count)
+        # Django QuerySet: run COUNT(*) in the parallel default thread pool.
+        # Django's acount() is sync_to_async(thread_sensitive=True) internally,
+        # which funnels every concurrent request through ONE shared thread.
+        if hasattr(queryset, "_iterable_class") and hasattr(queryset, "model"):
+            return await sync_to_thread(queryset.count)
+        # Non-Django async collections that expose acount
         if hasattr(queryset, "acount"):
             return await queryset.acount()
-        # Fallback to sync count wrapped in sync_to_async (for Django QuerySet)
+        # Other sync collections with a count() method
         elif hasattr(queryset, "count") and callable(queryset.count):
```

Ruff (0.15.21)

[warning] Unnecessary `elif` after `return` statement

Remove unnecessary `elif`

(RET505)